utils/CMakeLists.txt

```diff
@@ -15,3 +15,10 @@ file(TO_CMAKE_PATH "${LLVM_BUILD_BINARY_DIR}/bin/FileCheck${CMAKE_EXECUTABLE_SUF
 swift_install_in_component(PROGRAMS "${_SWIFT_UTILS_FILECHECK}"
                            DESTINATION bin
                            COMPONENT llvm-toolchain-dev-tools)
+
+# Copy over llvm-ar to use in building the Swift portions of the corelibs on
+# Unix platforms.
+file(COPY ${LLVM_RUNTIME_OUTPUT_INTDIR}/bin/llvm-ar
+  DESTINATION "${SWIFT_RUNTIME_OUTPUT_INTDIR}"
+  FILE_PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ
+  GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
```

Comment:
This really should be done outside of the CMake in the build-script python logic.  The way that we handle this on Windows is by using the LLVM distribution functionality.

Reply:
Any particular reason you think this should be in Python instead? I could see someone porting the toolchain to a new platform only by using CMake without `build-script`, as I did natively on Android initially, and this being in CMake would keep that working for new Unix platforms.

As for Windows, I simply disabled this copy on OS X and Windows in the latest version of this pull.